modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageServiceImpl.java

```diff
@@ -258,6 +275,20 @@ public Publisher<Entry> prefix(ByteArray prefix, long revUpperBound) {
         return new CursorPublisher(context, createPrefixCommand);
     }
 
+    /**
+     * Sends safe time sync message. Should be called only on the leader node.
+     *
+     * @param safeTime New safe time.
+     * @return Future that will be completed when message is sent.
+     */
+    public CompletableFuture<Void> syncTime(HybridTimestamp safeTime) {
```

Comment:
I noticed that you do not use the return value, would it not be a mistake to always send a new safeTime even if we did not process the previous one.

Reply:
See todo

Reply:
Where todo ?